ImportError: cannot import name 'gaussian' from 'scipy.signal'

e.g., https://github.com/ICB-DCM/pyPESTO/actions/runs/8534329173:

```
_________________ ERROR collecting test/sample/test_sample.py __________________
ImportError while importing test module '/home/runner/work/pyPESTO/pyPESTO/test/sample/test_sample.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/opt/hostedtoolcache/Python/3.9.19/x64/lib/python3.9/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
test/sample/test_sample.py:17: in <module>
    from pypesto.sample.pymc import PymcSampler
pypesto/sample/pymc.py:7: in <module>
    import arviz as az
.tox/base/lib/python3.9/site-packages/arviz/__init__.py:34: in <module>
    from .plots import *
.tox/base/lib/python3.9/site-packages/arviz/plots/__init__.py:2: in <module>
    from .autocorrplot import plot_autocorr
.tox/base/lib/python3.9/site-packages/arviz/plots/autocorrplot.py:7: in <module>
    from .plot_utils import default_grid, filter_plotters_list, get_plotting_function
.tox/base/lib/python3.9/site-packages/arviz/plots/plot_utils.py:15: in <module>
    from ..stats.density_utils import kde
.tox/base/lib/python3.9/site-packages/arviz/stats/__init__.py:3: in <module>
    from .density_utils import *
.tox/base/lib/python3.9/site-packages/arviz/stats/density_utils.py:8: in <module>
    from scipy.signal import convolve, convolve2d, gaussian  # pylint: disable=no-name-in-module
E   ImportError: cannot import name 'gaussian' from 'scipy.signal' (/home/runner/work/pyPESTO/pyPESTO/.tox/base/lib/python3.9/site-packages/scipy/signal/__init__.py)
```

Related to the recent [scipy release](https://github.com/scipy/scipy/releases/tag/v1.13.0)

See also https://github.com/arviz-devs/arviz/issues/2330

Nothing to be fixed in pypesto, but I guess others will run into this issue too...

--- a/setup.cfg
+++ b/setup.cfg
@@ -117,7 +117,6 @@ mpi =
     mpi4py >= 3.0.3
 pymc =
     arviz >= 0.12.1
-    scipy < 1.13.0 # https://github.com/ICB-DCM/pyPESTO/issues/1354
     aesara >= 2.8.6
     pymc >= 4.2.1
 aesara =

--- a/setup.cfg
+++ b/setup.cfg
@@ -117,7 +117,7 @@ mpi =
     mpi4py >= 3.0.3
 pymc =
     arviz >= 0.12.1
-    scipy < 1.13.0 # https://github.com/ICB-DCM/pyPESTO/issues/1354
+    scipy < 1.13.0; python_version=='3.9' # https://github.com/ICB-DCM/pyPESTO/issues/1354
     aesara >= 2.8.6
     pymc >= 4.2.1
 aesara =

--- a/setup.cfg
+++ b/setup.cfg
@@ -117,6 +117,7 @@ mpi =
     mpi4py >= 3.0.3
 pymc =
     arviz >= 0.12.1
+    scipy < 1.13.0 # https://github.com/ICB-DCM/pyPESTO/issues/1354
     aesara >= 2.8.6
     pymc >= 4.2.1
 aesara =

